src/General/Errors.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE DeriveDataTypeable, PatternGuards, RecordWildCards, ConstraintKinds #-}
+
+-- | Errors seen by the user
+module General.Errors(
```

Comment:
I'd get rid of this module entirely too. Just use `error` wherever you were throwing an error. It's only a small number of places.